yaml/build.go

```diff
@@ -63,22 +65,31 @@ func handleBuildArgs(value interface{}) (map[string]string, error) {
 	case map[interface{}]interface{}:
 		return handleBuildArgMap(v)
 	case []interface{}:
-		args = map[string]string{}
-		for _, elt := range v {
-			uniqArgs, err := handleBuildArgs(elt)
-			if err != nil {
-				return args, nil
-			}
-			for mapKey, mapValue := range uniqArgs {
-				args[mapKey] = mapValue
-			}
-		}
-		return args, nil
+		return handleBuildArgSlice(v)
 	default:
 		return args, fmt.Errorf("Failed to unmarshal Build args: %#v", value)
 	}
 }
 
+func handleBuildArgSlice(s []interface{}) (map[string]string, error) {
+	var args = map[string]string{}
+	for _, arg := range s {
+		// check if a value is provided
+		switch v := strings.SplitN(arg.(string), "=", 2); len(v) {
+		case 1:
+			// we don't use EnvironmentLookup here as docker does not use values in the env_file or environment config stanza for this
+			envValue := os.Getenv(v[0])
```

Comment:
Ok.  I've updated things so that the main processing is still done in yaml/build.go, but it sets empty strings as map values (indicated by an ASCII null)  if the build Arg *value* is not provided in the compose file.   Then, when the service is created it will use ```EnvironmentLookup``` to see if there is a value for that Arg available in the environment.  I think it works nicely to delay looking for the value in the environment until the service is created as you may want to take some action between reading in the compose file and starting the services.